Python/instrumentation.c

```diff
@@ -1577,6 +1551,20 @@ _Py_Instrument(PyCodeObject *code, PyInterpreterState *interp)
         return 0;
     }
     int code_len = (int)Py_SIZE(code);
+    if (code->co_executors != NULL && code->co_executors->size > 0 ) {
+        for (int i = 0; i < code_len; i += _PyInstruction_GetLength(code, i)) {
+            _Py_CODEUNIT *instr = &_PyCode_CODE(code)[i];
+            uint8_t opcode = instr->op.code;
+            uint8_t oparg = instr->op.arg;
+            if (opcode == ENTER_EXECUTOR) {
+                _PyExecutorObject *exec = code->co_executors->executors[oparg];
+                assert(exec->vm_data.opcode != ENTER_EXECUTOR);
+                instr->op.code = exec->vm_data.opcode;
+                instr->op.arg = exec->vm_data.oparg;
+            }
+        }
```

Comment:
Can you move this loop into `_PyCode_Clear_Executors()`? Also it should be done whenever `code->co_executors != NULL`, regardless of `code->co_executors->size`.